src/Shared/Debugger/HttpContextDebugFormatter.cs

```diff
@@ -0,0 +1,41 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+using System.Globalization;
+using Microsoft.AspNetCore.Http;
+using Microsoft.AspNetCore.Http.Features;
+
+namespace Microsoft.AspNetCore.Shared;
+
+internal static class HttpContextDebugFormatter
+{
+    public static string ResponseToString(HttpResponse response, string? reasonPhrase)
+    {
+        var text = response.StatusCode.ToString(CultureInfo.InvariantCulture);
+        var resolvedReasonPhrase = response.HttpContext.Features.Get<IHttpResponseFeature>()?.ReasonPhrase ?? reasonPhrase;
+        if (!string.IsNullOrEmpty(resolvedReasonPhrase))
+        {
+            text += $" {resolvedReasonPhrase}";
+        }
+        if (!string.IsNullOrEmpty(response.ContentType))
+        {
+            text += $" {response.ContentType}";
+        }
+        return text;
+    }
+
+    public static string RequestToString(HttpRequest request, bool includeQueryString)
+    {
+        var text = $"{request.Method} {request.Scheme}://{request.Host.Value}{request.PathBase.Value}{request.Path.Value}{(includeQueryString ? request.QueryString.Value : string.Empty)} {request.Protocol}";
+        if (!string.IsNullOrEmpty(request.ContentType))
+        {
+            text += $" {request.ContentType}";
+        }
+        return text;
+    }
+
+    public static string ContextToString(HttpContext context, string? reasonPhrase)
+    {
+        return $"Request = {RequestToString(context.Request, includeQueryString: false)}, Response = {ResponseToString(context.Response, reasonPhrase)}";
```

Comment:
It's still going to be too much, paths and hosts are often long too. I think the `Request =` markers are taking up valuable space.
The key pieces of information are:
- Method
- Path
- Status code (& description?)
` POST /identity/users/sign-in 401 Unauthorized`

Reply:
Hmm, I think knowing HTTP vs HTTPS and host and port is still useful. The host name usually isn't that long (a lot of the time it will be localhost)

Reply:
Updated. I used your format but included the scheme/host/port.

![image](https://github.com/dotnet/aspnetcore/assets/303201/bbb429a5-045a-4c5e-812a-1b1876aeae53)

I think it balances displaying useful information vs size. If someone does have a very long path or host then they'll have to expand to view more detail, but I don't think we should hinder the usual experience for them.

Reply:
I'd mainly excluded the scheme, host, and port because they're less likely to be different across requests, but this is a reasonable compromise.

Reply:
Done.